waflib/Tools/ccroot.py

```diff
@@ -792,3 +792,12 @@ def set_full_paths_hpux(self):
 			else:
 				lst.append(os.path.normpath(os.path.join(base, x)))
 		self.env[var] = lst
+
+def pdb_flag(self):
+	if '/Zi' not in self.zi_env_flags():
```

Comment:
I would probably just encapsulate the logic here

if '/Zi' in self.env.CXXFLAGS or self.env.CFLAGS and drop the requirement to have a subclass.

self.env.get('CXXFLAGS', []) if necessary